### ISSUE TITLE
SLSTM cuda/cpu/mps bug

### ISSUE THREAD

**Issue by jeshraghian** [jeshraghian/snntorch#198]
Only `cuda` and `cpu` are accounted for as devices types in SLSTM.

See line 295 of `slstm.py`:

```
def _reshape_input(self, input_):
        if input_.is_cuda:
            device = "cuda"
        else:
            device = "cpu"
        b, _ = input_.size()
        return torch.zeros(b, self.hidden_size).to(device)
```

**Comment by jeshraghian** [comment]
Fix:

```
def _reshape_input(self, input_):
        device = input_.device
        b, _ = input_.size()
        return torch.zeros(b, self.hidden_size).to(device)
```

To-do: Check if SConv2dLSTM has the same bug.

**Comment by jeshraghian** [comment]
Whoops, I fixed this 3 weeks ago in the v0.6.1 update, but forgot to update it locally...
